src/dataExplorer/components/DataExplorerPage.tsx

```diff
@@ -137,7 +137,7 @@ const DataExplorerPage: FC = () => {
     (scriptQueryBuilder && isFlagEnabled('newDataExplorer')) || isNewIOxOrg
 
   const shouldShowSaveAsButton =
-    !isNewIOxOrg ||
+    !useSelector(selectIsNewIOxOrg) ||
     shouldShowNotebooks ||
```

Comment:
This edit is a minor bug fix that requires some explanation because it's pretty far down the logic tree of what can happen within the matrix of feature overrides.

`shouldShowSaveAsButton` controls whether the "Save As" button on data explorer displays. This variable evaluates to `true` and permits the display of the "Save As" button in data explorer if the user is _not_ in an IOx org; or, if the user _is_ in an IOx org, but still has access to any one of the following: notebooks, tasks, dashboards, or variables. That can happen if the user is in the IOx lighthouse program, or has a feature override flag enabled. 

The logic wasn't right here for a specific edge case that appeared during the fixed e2e tests:
1. You're in an IOx org and
2. `showOldDataExplorerInNewIOx` is on and
3. All other overrides are off

In that case, `!isNewIOxOrg` incorrectly evaluates to `true`, since the override flag for data explorer is _not_ off (`!isFlagEnabled('showOldDataExplorerInNewIOx')`). This shows the "Save As" button even though the resulting overlay wouldn't offer you any place to save the query. The "Save As" button shouldn't care whether you have the `showOldDataExplorerInNewIOx` flag on or off. The "Save As" button should only display if you're in a TSM/IOx lighthouse org; or one of the tasks/notebooks/dashboards/variables overrides is on.

Reply:
We are getting some weird behaviors, because we have conditionals (including flags and iox org checking) at multiple levels in our code. In the scripts tests, we ran into a conflict between the `isFlagEnabled('showOldDataExplorerInNewIOx')` and the AppSettings global state of `scriptQueryBuilder`.

We are merging in a new fixture with [this PR ](https://github.com/influxdata/ui/pull/6593#discussion_r1099621263)(hopefully merged today), which will have a new `cy.mockIsCloud2Org()` which returns an older createdAt date for the org.

This means that any deprecated features based on org createdAt date, will be turned back on if we use this in the beforeEach(). This includes features deprecated on iox (e.g. marty release), and also in the future could be used on tsm if we deprecate features prior to the migration to iox backend for those folks.

Reply:
People now have the choice of using the flags (e.g. `showDashboardsInNewIOx`) as needed for some tests. But for features that have nested conditionals throughout shared logic (a.k.a. old and new data explorer), using this `cy.mockIsCloud2Org()`, followed by `cy.switchToDataExplorer('old')`may be easier.